prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/FHIRBundleHelpersTests.kt

```diff
@@ -0,0 +1,344 @@
+package gov.cdc.prime.router.fhirengine.engine.fhirRouterTests
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isFalse
+import assertk.assertions.isNotEmpty
+import assertk.assertions.isNotNull
+import assertk.assertions.isNull
+import assertk.assertions.isTrue
+import gov.cdc.prime.router.ActionLogger
+import gov.cdc.prime.router.CustomerStatus
+import gov.cdc.prime.router.DeepOrganization
+import gov.cdc.prime.router.Metadata
+import gov.cdc.prime.router.Organization
+import gov.cdc.prime.router.Receiver
+import gov.cdc.prime.router.Schema
+import gov.cdc.prime.router.Topic
+import gov.cdc.prime.router.azure.BlobAccess
+import gov.cdc.prime.router.azure.DatabaseAccess
+import gov.cdc.prime.router.azure.QueueAccess
+import gov.cdc.prime.router.fhirengine.engine.RawSubmission
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.CustomContext
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.FhirPathUtils
+import gov.cdc.prime.router.fhirengine.utils.FHIRBundleHelpers
+import gov.cdc.prime.router.fhirengine.utils.FHIRBundleHelpers.deleteResource
+import gov.cdc.prime.router.fhirengine.utils.FHIRBundleHelpers.getResourceProperties
+import gov.cdc.prime.router.fhirengine.utils.FHIRBundleHelpers.getResourceReferences
+import gov.cdc.prime.router.fhirengine.utils.FhirTranscoder
+import io.mockk.clearAllMocks
+import io.mockk.mockkClass
+import io.mockk.spyk
+import org.hl7.fhir.r4.model.Endpoint
+import org.hl7.fhir.r4.model.Observation
+import org.hl7.fhir.r4.model.Provenance
+import org.jooq.tools.jdbc.MockConnection
+import org.jooq.tools.jdbc.MockDataProvider
+import org.jooq.tools.jdbc.MockResult
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.TestInstance
+import java.io.File
+import java.lang.IllegalStateException
+import java.util.UUID
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class FHIRBundleHelpersTests {
```

Comment:
Should we have tests that delete DiagnosticReport resource and one other type of resource?

Reply:
Added a test that removes a Provenance resource

Reply:
@victor-chaparro Provenance is not too complicated. What would happen if you delete a Diagnostic Report which has a lot of stuff?

Reply:
@carlosfelix2 there is one example that deletes a diagnostic repot after deleting the observation, but most of what's referenced in that diagnostic report is referenced in other diagnostic reports, so not much happens. Do you have an example of one bundle with just one Diagnostic Report? If not I'll try to come up with one.